src/utils.js

```diff
@@ -343,3 +348,31 @@ const sleep = (millis) => {
 export const publicUtils = {
     sleep,
 };
+
+/**
+ * Returns a promise that resolves to an array of urls parsed from the resource
+ * available at the provided url.
+ *
+ * @param {String} url
+ * @param {String} encoding
+ * @param {RegExp} urlRegExp
+ * @returns {Promise<Array>}
+ */
+export const downloadListOfUrls = ({ url, encoding, urlRegExp }) => {
+    if (!url) return Promise.reject(new Error('Resource URL must be provided.'));
+    encoding = encoding || 'utf8';
```

Comment:
I would do this differently:

```
export const downloadListOfUrls = ({ url, encoding = 'utf8', urlRegExp = URL_NO_COMMAS_REGEX }) => {
    checkParamOrThrow(url, 'url', 'String');
    checkParamOrThrow(encoding, 'encoding', 'String');
    checkParamOrThrow(urlRegExp, 'urlRegExp', 'Object');

   ...
};
```

(1) Because this way for encoding user has to enter the string and cannot enter for example null or false or 0 that would unexpectedly fallback to 'utf8' and also it's more obvious from code what are the defaults.

(2) We should provide some default for regex that should be documented here and in UrlList.